common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -93,12 +94,13 @@ public static CalendarInterval fromString(String s) {
   }
 
   /**
-   * Convert a string to CalendarInterval. Unlike fromString, this method is case-insensitive and
-   * will throw IllegalArgumentException when the input string is not a valid interval.
+   * Convert a string to CalendarInterval. Unlike fromString, this method can handle
+   * strings without the `interval` prefix and throws IllegalArgumentException
+   * when the input string is not a valid interval.
    *
    * @throws IllegalArgumentException if the string is not a valid internal.
    */
-  public static CalendarInterval fromCaseInsensitiveString(String s) {
+  public static CalendarInterval fromStringWithOptionalPrefix(String s) {
```

Comment:
I'd like to recommend you not to rename this. The updated function description is enough for me.

Reply:
ok. I will revert this renaming.